pytorch_lightning/core/lightning.py

```diff
@@ -390,7 +390,11 @@ def training_step(self, *args, **kwargs):
                 :paramref:`~pytorch_lightning.trainer.trainer.Trainer.truncated_bptt_steps` > 0.
 
         Return:
-            roch.Tensor or a dictionary with anything you want (must include the keyword 'loss')
+            Any of.
+
+            - `torch.Tensor`
+            - `dict` - Can include any keys, but must include the key 'loss'
+            - `None` - Training will skip to the next batch
```

Comment:
can the typehint be added for this

```
def training_step(self, *args, **kwargs) -> Optional[Union[torch.Tensor, Dict[Any, Any]]
```

Reply:
Mind use suggestions, it is easier to integrate...

Reply:
but type hints are so annoying when autocompleting on IDEs lol. 

all this boilerplate pastes and then you have to delete it or import the typing stuff.

Reply:
So... yay or nay for adding hints?

Reply:
IMO we should definitely add them, since this really helps users to keep track of the types (and probably devs as well)

Reply:
Done!